lnutil/dlclib.go

```diff
@@ -0,0 +1,481 @@
+	}
+
+	opArr := OutPointToBytes(self.FundingOutpoint)
+	buf.Write(opArr[:])
+
+	return buf.Bytes()
+}
+
+// GetDivision loops over all division specifications inside the contract and returns the one matching the requested oracle value
+func (c DlcContract) GetDivision(value int64) (*DlcContractDivision, error) {
+	for _, d := range c.Division {
+		if d.OracleValue == value {
+			return &d, nil
+		}
+	}
+
+	return nil, fmt.Errorf("Division not found in contract")
+}
+
+// GetTheirSettlementSignature loops over all stored settlement signatures from the counter party and
+// returns the one matching the requested oracle value
+func (c DlcContract) GetTheirSettlementSignature(value int64) ([64]byte, error) {
+
+	for _, s := range c.TheirSettlementSignatures {
+		if s.Outcome == value {
+			return s.Signature, nil
+		}
+	}
+
+	return [64]byte{}, fmt.Errorf("Signature not found in contract")
+}
+
+// PrintTx prints out a transaction as serialized byte array to StdOut
+func PrintTx(tx *wire.MsgTx) {
+	var buf bytes.Buffer
+	w := bufio.NewWriter(&buf)
+	tx.Serialize(w)
+	w.Flush()
+	fmt.Printf("%x\n", buf.Bytes())
+}
+
+// DlcOutput returns a Txo for a particular value that pays to (PubKeyPeer+PubKeyOracleSig or (OurPubKey and TimeDelay))
+func DlcOutput(pubKeyPeer, pubKeyOracleSig, ourPubKey [33]byte, value int64) *wire.TxOut {
+	scriptBytes := DlcCommitScript(pubKeyPeer, pubKeyOracleSig, ourPubKey, 5)
+	scriptBytes = P2WSHify(scriptBytes)
+
+	return wire.NewTxOut(value, scriptBytes)
+}
+
+// DLC Commit script makes a script that pays to (PubKeyPeer+PubKeyOracleSig or (OurPubKey and TimeDelay))
+// We send this over (signed) to the other side. If they publish the TX with the correct script they can use
+// The oracle's signature and their own private key to claim the funds from the output. However,
+// If they send the wrong one, they won't be able to claim the funds - and we can claim them once the
+// Time delay has passed.
+func DlcCommitScript(pubKeyPeer, pubKeyOracleSig, ourPubKey [33]byte, delay uint16) []byte {
+	builder := txscript.NewScriptBuilder()
+
+	// 1 for penalty / revoked, 0 for timeout
+	// 1, so timeout
+	builder.AddOp(txscript.OP_IF)
+
+	// Combine pubKey and Oracle Sig
+	combinedPubKey := CombinePubs(pubKeyPeer, pubKeyOracleSig)
+
+	builder.AddData(combinedPubKey[:])
+
+	// 0, so revoked
+	builder.AddOp(txscript.OP_ELSE)
+
+	// CSV delay
+	builder.AddInt64(int64(delay))
+	// CSV check, fails here if too early
+	builder.AddOp(txscript.OP_NOP3) // really OP_CHECKSEQUENCEVERIFY
+	// Drop delay value
+	builder.AddOp(txscript.OP_DROP)
+	// push timeout key
+	builder.AddData(ourPubKey[:])
+
+	builder.AddOp(txscript.OP_ENDIF)
+
+	// check whatever pubkey is left on the stack
+	builder.AddOp(txscript.OP_CHECKSIG)
+
+	// never any errors we care about here.
+	s, _ := builder.Script()
+
+	return s
+}
+
+// BigIntToEncodedBytes converts a big integer into its corresponding
+// 32 byte little endian representation.
```

Comment:
sure this is little endian? I think it's big..?

Reply:
Also it's quite possible that *I* wrote this and it was copied here (I sortof vaguely remember writing something like this to pad out s values) so I might have written that it's little endian instead of big.  Pretty sure bigint.Bytes() returns big endian though.

Reply:
I copied it from https://github.com/adiabat/dlctest/blob/master/sig.go#L45

Reply:
yup, big endian (ref. https://golang.org/pkg/math/big/#Int.Bytes)